samza-core/src/main/java/org/apache/samza/serializers/model/JsonContainerModelMixIn.java

```diff
@@ -19,23 +19,28 @@
 
 package org.apache.samza.serializers.model;
 
-import java.util.Map;
 import org.apache.samza.container.TaskName;
 import org.apache.samza.job.model.TaskModel;
 import org.codehaus.jackson.annotate.JsonCreator;
 import org.codehaus.jackson.annotate.JsonProperty;
 
+import java.util.Map;
+
 /**
  * A mix-in Jackson class to convert Samza's ContainerModel to/from JSON.
  */
 public abstract class JsonContainerModelMixIn {
   @JsonCreator
-  public JsonContainerModelMixIn(@JsonProperty("container-id") int containerId, @JsonProperty("tasks") Map<TaskName, TaskModel> tasks) {
+  public JsonContainerModelMixIn(@JsonProperty("processor-id") String processorId, @JsonProperty("tasks") Map<TaskName, TaskModel> tasks) {
```

Comment:
When we read the old JSON container model, we won't be using the mixin anymore. I changed it to use the de-serializer. If the containerModel contains processor-id, all is well. If it  doesn't contain processor-id, it will return null. We, then, check to see if container-id is defined. If it is, it is converted to a processor-id String. Otherwise, we throw an exception as we expect the Container Model to either contain container-id or processor-id. 
I was able to verify the deserialization with a unit test in TestSamzaObjectMapper. 
Let me know if you are talking about something else here.